### ISSUE TITLE
Is autotune using the correct values for Loop users with Omnipod

### ISSUE THREAD

**Issue by bartsluimer** [openaps/oref0#1362]
Loop with Omnipod is using a different timing with basal as Medtronic. This is causing a big difference in for example the day to day report inside Nightscout. 
The question is what values is autotune using. 

Example on difference between bolus and basal wit Tidepool and Nightscout:
![64612394-f85e4a80-d3d3-11e9-996d-2fe064bd0280](https://user-images.githubusercontent.com/25899216/74513020-e1370200-4f09-11ea-9cc0-f4aa67ccaa51.png)
![64612213-8c7be200-d3d3-11e9-9c30-ebacb61f3e15](https://user-images.githubusercontent.com/25899216/74513031-e5631f80-4f09-11ea-90ed-1d31ae174ecf.png)

Tidepool values are correct.

Loop is now sending different values, to prevent any mistakes. Only is Autotune using these new timing and amount value?:

**Old TempBasal**

{
"_id": {
"$oid": "5d53811e7cc3d56f0c3fe930"
},
"timestamp": "2019-08-14T03:28:47Z",
"eventType": "Temp Basal",
"enteredBy": "loop://Marion iPhone 7+",
"duration": 4,
"temp": "absolute",
"absolute": 2.55,
"rate": 2.55,
"created_at": "2019-08-14T03:28:47.000Z",
"utcOffset": 0
}

**New TempBasal**

{
"_id": {
"$oid": "5d792b1f7cc3d56f0c739952"
},
"absolute": 1.75,
"rate": 1.75,
"temp": "absolute",
"duration": 4.767875816424688,
"timestamp": "2019-09-11T17:12:57Z",
"enteredBy": "loop://Marion iPhone 7+",
"eventType": "Temp Basal",
"amount": 0.1,
"created_at": "2019-09-11T17:12:57.000Z",
"utcOffset": 0
}

**Comment by scottleibrand** [comment]
The easiest way to test this would be using the NS URL of someone using Omnipod Loop. Do we have anyone willing to disclose their NS URL for that purpose?

**Comment by bartsluimer** [comment]
Will delete the url once finished. https://amklaassen.herokuapp.com/

**Comment by sethgagnon** [comment]
Thanks @bartsluimer for opening. So, this is related to this Issue then as well:
https://github.com/nightscout/cgm-remote-monitor/issues/4971

**Comment by bartsluimer** [comment]
That is correct. The issue became visible when comparing Nightscout day to day report vs Tidepool data. Tidepool is just using Apple health real values. Nightscout was using a different approach. 

So basically when using omnipod en Loop people do show wrong day to day reports to their health team.

Regarding autotune we were doubting if the correct values are used. A lot of people with omnipod and Loop do not experience a lot of benefit in using autotune.

**Comment by sethgagnon** [comment]
Happy to share my NS site as well, although I think @scottleibrand may already have it from an issue or two last year :)

If not, let me know and I can pass it along. I would love to ensure Autotune is using the proper values. Thanks!!

**Comment by scottleibrand** [comment]
I checked and confirmed that autotune does not directly use the new `duration` and `amount` fields that Loop adds to Temp Basal records. I haven't modeled exactly how it would be affected by the Omnipod pulse timing, and whether it would match what's shown in Nightscout, but lib/iob/history.js and possibly other files will need to be updated to use those new fields to accurately calculate IOB from Omnipod Loops.

**Comment by scottleibrand** [comment]
Possible fix for testing: https://github.com/openaps/oref0/pull/1364

**Comment by sethgagnon** [comment]
Just tried out the fix, figured I would share some before and after runs for 1 day:

Here is the output from dev branch:
Parameter      | Pump        | Autotune    | Days Missing
---------------------------------------------------------
ISF [mg/dL/U]  | 60.000      | 58.224      |
Carb Ratio[g/U]| 8.100       | 7.840       |
  00:00        | 1.250       | 1.270       | 1           
  01:00        | 1.250       | 1.305       | 1           
  02:00        | 1.250       | 1.327       | 1           
  03:00        | 1.250       | 1.359       | 0           
  04:00        | 1.150       | 1.216       | 0           
  05:00        | 1.150       | 1.196       | 0           
  06:00        | 1.200       | 1.218       | 1           
  07:00        | 1.250       | 1.258       | 1           
  08:00        | 1.300       | 1.287       | 2           
  09:00        | 1.250       | 1.255       | 2           
  10:00        | 1.250       | 1.255       | 2           
  11:00        | 1.250       | 1.255       | 2           
  12:00        | 1.250       | 1.255       | 2           
  13:00        | 1.300       | 1.295       | 1           
  14:00        | 1.300       | 1.327       | 1           
  15:00        | 1.300       | 1.343       | 1           
  16:00        | 1.300       | 1.338       | 1           
  17:00        | 1.300       | 1.324       | 1           
  18:00        | 1.350       | 1.374       | 1           
  19:00        | 1.350       | 1.374       | 1           
  20:00        | 1.350       | 1.356       | 1           
  21:00        | 1.350       | 1.327       | 1           
  22:00        | 1.300       | 1.287       | 1           
  23:00        | 1.300       | 1.290       | 2         

Here is the output using the fix in 1364 (omnipod-iob branch):
Parameter      | Pump        | Autotune    | Days Missing
---------------------------------------------------------
ISF [mg/dL/U]  | 60.000      | 58.000      |
Carb Ratio[g/U]| 8.100       | 7.933       |
  00:00        | 1.250       | 1.244       | 1           
  01:00        | 1.250       | 1.243       | 1           
  02:00        | 1.250       | 1.240       | 1           
  03:00        | 1.250       | 1.230       | 2           
  04:00        | 1.150       | 1.136       | 0           
  05:00        | 1.150       | 1.139       | 0           
  06:00        | 1.200       | 1.189       | 0           
  07:00        | 1.250       | 1.251       | 1           
  08:00        | 1.300       | 1.286       | 2           
  09:00        | 1.250       | 1.254       | 2           
  10:00        | 1.250       | 1.254       | 2           
  11:00        | 1.250       | 1.254       | 2           
  12:00        | 1.250       | 1.254       | 2           
  13:00        | 1.300       | 1.295       | 1           
  14:00        | 1.300       | 1.327       | 1           
  15:00        | 1.300       | 1.343       | 1           
  16:00        | 1.300       | 1.337       | 1           
  17:00        | 1.300       | 1.323       | 1           
  18:00        | 1.350       | 1.374       | 1           
  19:00        | 1.350       | 1.374       | 1           
  20:00        | 1.350       | 1.355       | 1           
  21:00        | 1.350       | 1.327       | 1           
  22:00        | 1.300       | 1.287       | 1           
  23:00        | 1.300       | 1.289       | 2

**Comment by bartsluimer** [comment]
@sethgagnon which Loop branch are you currently using? Assume that the autobolus branch will cause less difference. Everything more as basal is bolus.

**Comment by sethgagnon** [comment]
I am using the AB branch

**Comment by bartsluimer** [comment]
@sethgagnon that is probably having less difference. @scottleibrand what kind of testing is required? Shall we ask the Loop community for assistance?

**Comment by scottleibrand** [comment]
Once a few people have run autotune using this branch over long enough timeframes and are seeing reasonable results, I'll be happy to merge it to dev.

**Comment by bartsluimer** [comment]
Will ask on Zulip for assistance. Currently our values are to far off to assist. Nightscout is from my wife and we just had a baby.
